dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsStatusReportTask.java

```diff
@@ -87,35 +98,69 @@ abstract static class HdfsNameNodeStatus {
     @JsonProperty
     abstract ECBlockGroupStats ecBlockGroupStats();
 
-    public static HdfsNameNodeStatus create(ClientProtocol namenode) throws IOException {
+    public static HdfsNameNodeStatus create(
+        ReplicatedBlockStats replicatedBlockStats, ECBlockGroupStats ecBlockGroupStats) {
       return new AutoValue_HdfsStatusReportTask_HdfsNameNodeStatus(
-          namenode.getReplicatedBlockStats(), namenode.getECBlockGroupStats());
+          replicatedBlockStats, ecBlockGroupStats);
     }
   }
 
   @AutoValue
   abstract static class HdfsDataNodesStatus {
-    @JsonProperty
+    @JsonProperty("regular")
     abstract ImmutableList<DatanodeInfo> regularNodes();
 
-    @JsonProperty
+    @JsonProperty("slow")
     abstract ImmutableList<DatanodeInfo> slowNodes();
 
-    public static HdfsDataNodesStatus create(DistributedFileSystem dfs) throws IOException {
+    public static HdfsDataNodesStatus create(
+        List<DatanodeInfo> regularNodes, List<DatanodeInfo> slowNodes) {
       return new AutoValue_HdfsStatusReportTask_HdfsDataNodesStatus(
-          ImmutableList.copyOf(dfs.getDataNodeStats()),
-          ImmutableList.copyOf(dfs.getSlowDatanodeStats()));
+          ImmutableList.copyOf(regularNodes), ImmutableList.copyOf(slowNodes));
     }
   }
 
   @Override
   protected Void doRun(TaskRunContext context, @Nonnull ByteSink sink, @Nonnull Handle handle)
       throws IOException {
     DistributedFileSystem dfs = ((HdfsHandle) handle).getDfs();
-    HdfsStatus status = HdfsStatus.create(dfs);
+    HdfsStatus status = extractHdfsStatus(dfs);
     try (final Writer output = sink.asCharSink(UTF_8).openBufferedStream()) {
       OBJECT_MAPPER.writeValue(output, status);
     }
     return null;
   }
+
+  private HdfsStatus extractHdfsStatus(DistributedFileSystem dfs) throws IOException {
+    return HdfsStatus.create(
+        dfs.getStatus(),
+        dfs.setSafeMode(SafeModeAction.GET),
+        dfs.getBytesWithFutureGenerationStamps(),
+        extractNameNode(dfs.getClient().getNamenode()),
+        extractDataNodes(dfs));
+  }
+
+  private HdfsNameNodeStatus extractNameNode(ClientProtocol namenode) throws IOException {
+    return HdfsNameNodeStatus.create(
+        namenode.getReplicatedBlockStats(), namenode.getECBlockGroupStats());
+  }
+
+  private HdfsDataNodesStatus extractDataNodes(DistributedFileSystem dfs) {
+    ImmutableList<DatanodeInfo> regularNodes;
+    ImmutableList<DatanodeInfo> slowNodes;
+    try {
+      regularNodes = ImmutableList.copyOf(dfs.getDataNodeStats());
+    } catch (Exception ex) {
+      LOG.warn("Error retrieving data node status.", ex);
```

Comment:
Possible typo: the message refers to "status", but in the getter name it's "stats"

Reply:
Thanks. Fixed.